src/shared/api.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Customization, Customizations } from '../codewhisperer/client/codewhispereruserclient'
+import {
+    getAvailableCustomizationsList,
+    getSelectedCustomization,
+    setSelectedCustomization,
+} from '../codewhisperer/util/customizationUtil'
+
+export interface AwsToolkitApi {
+    setCustomization(customization: Customization): Promise<boolean>
+    getSelectedCustomization(): Customization
+    listAvailableCustomizations(): Promise<Customizations>
```

Comment:
To make it very clear that these types are public (and thus need backwards-compatibility), can we move the type/interface definitions (for `Customization`) to this file. That should help avoid a developer accidentally changing it in the future, in a breaking way.

If internal code needs a different `Customization` it can subclass the one that lives here.